packages/core/content-releases/server/src/register.ts

```diff
@@ -10,5 +12,17 @@ export const register = async ({ strapi }: { strapi: LoadedStrapi }) => {
     strapi.features.future.isEnabled('contentReleases')
   ) {
     await strapi.admin.services.permission.actionProvider.registerMany(ACTIONS);
+
+    const releaseActionService = getService('release-action', { strapi });
+    // Clean up release-actions when draft and publish is disabled
+    strapi.eventHub.on('content-type.update', async ({ contentType }) => {
+      if (contentType.schema.options.draftAndPublish === false) {
+        await releaseActionService.deleteManyForContentType(contentType.uid);
+      }
+    });
+    // Clean up release-actions when a content-type is deleted
+    strapi.eventHub.on('content-type.delete', async ({ contentType }) => {
+      await releaseActionService.deleteManyForContentType(contentType.uid);
+    });
```

Comment:
Ideally you'd [remove](https://contributor.strapi.io/docs/core/strapi/event-hub#off) the listener during the `destroy` lifecycle. And if you do that it would be more convenient to have just 1 thing to remove, therefore to use `subscribe` instead of `on`

Reply:
oh right good point forgot about that.

Reply:
I've done the setup to destroy my listeners but I think I'm a bit confused. Is it necessary seeing as though Strapi removes all listeners via eventHub.destroy() after the plugin lifecycle functions are run?

https://github.com/strapi/strapi/blob/fa02f6c7db1fdd4246f822ef54b4e7c56c1e377e/packages/core/strapi/src/Strapi.ts#L354-L369

Reply:
I see what you mean about having just 1 thing to remove but using subscribe wasn't adding up in my head since it will listen to _every_ event and I liked the idea of being specific about what I want to listen to. 

I think you have a better understanding of the eventHub though so let me know if you think it doesn't make sense what I've done here.